W-18609692 prompt decorator gr (#858)
mulesoft/docs-gateway: gateway/1.9/modules/ROOT/pages/policies-included-a2a-prompt-decorator.adoc

>s|Summary      | Decorates prompts with context information
>s|Category | A2A
>s|First Flex Gateway version available | v1.9.3
>s| Returned Status Codes | No return codes exist for this policy.
|===

== Summary

> The A2A Prompt Decorator policy injects additional content into the user provided prompt. This enables you to provide custom prompts you want your upstream server to receive. Configure the text you want to inject as either a DataWeave string parameter or as a file containing the text you want to inject. You can filter what requests you want to inject prompts into by using a DataWeave expression.

Reviewer: ```suggestion
The A2A Prompt Decorator policy injects additional content into the user-provided prompt. This enables you to provide custom prompts you want your upstream server to receive. Configure the text you want to inject as either a DataWeave string parameter or as a file containing the text you want to inject. You can filter what requests you want to inject prompts into by using a DataWeave expression.
```

> The A2A Prompt Decorator policy injects additional content into the user provided prompt. This enables you to provide custom prompts you want your upstream server to receive. Configure the text you want to inject as either a DataWeave string parameter or as a file containing the text you want to inject. You can filter what requests you want to inject prompts into by using a DataWeave expression.

Reviewer: ```suggestion
The A2A Prompt Decorator policy injects additional content into provided prompts before they are sent to the upstream server. You can specify the content to inject either as a DataWeave string parameter or by referencing a file. Additionally, you can use a DataWeave expression to filter which requests should have prompts decorated.
```

== Configuring Policy Parameters

include::partial$policy-title-headers.adoc[tag=configFileTitleOnly]

The A2A Prompt Decorator policy is not supported in Local Mode.

include::partial$policy-title-headers.adoc[tag=ui]

[%header%autowidth.spread,cols="a,a"]

|===
| Element | Description
| Text Decorators | List of text decorations you want to add to the prompts.
| Prompt Text | DataWeave Expression that generates the prompt text injected into the request. For example: `#['Use Imperial metric units']`.
| Condition | DataWeave Expression that filters which prompts to decorate. For example: `#[attributes.headers['id']==1]`.

> | File Decorators | File Decorators

Reviewer: ```suggestion
| File Decorators | List of file decorations to add to the prompts.
```
